ddtrace/tracer/startup.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package tracer
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+)
+
+type startupInfo struct {
+	// Common startup status
+	Date                  string                 `json:"date"`
+	OSName                string                 `json:"os_name"`
+	OSVersion             string                 `json:"os_version"`
+	Version               string                 `json:"version"`
+	Lang                  string                 `json:"lang"`
+	LangVersion           string                 `json:"lang_version"`
+	Env                   string                 `json:"env"`
+	Service               string                 `json:"service"`
+	AgentURL              string                 `json:"agent_url"`
+	AgentReachable        bool                   `json:"agent_reachable"`
+	AgentError            error                  `json:"agent_error"`
+	Debug                 bool                   `json:"debug"`
+	AnalyticsEnabled      bool                   `json:"analytics_enabled"`
+	SampleRate            float64                `json:"sample_rate"`
+	SamplingRules         []SamplingRule         `json:"sampling_rules"`
+	SamplingRulesError    error                  `json:"sampling_rules_error"`
+	Tags                  map[string]interface{} `json:"tags"`
+	RuntimeMetricsEnabled bool                   `json:"runtime_metrics_enabled"`
+
+	//Go-tracer-specific startup status
+	GlobalService string `json:"global_service"`
+}
+
+func agentReachable(t *tracer) (bool, error) {
+	req, err := http.NewRequest("POST", fmt.Sprintf("http://%s/v0.4/traces", resolveAddr(t.config.agentAddr)), strings.NewReader("[]"))
+	if err != nil {
+		return false, fmt.Errorf("cannot create http request: %v", err)
+	}
+
+	req.Header.Set(traceCountHeader, "0")
+	req.Header.Set("Content-Length", "2")
+	c := &http.Client{}
+	response, err := c.Do(req)
```

Comment:
```suggestion
	response, err := defaultClient.Do(req)
```
We have it in `transport.go`